### PR TITLE
sql/migratespec: support modify_table and add_column

### PR DESCRIPTION
**Pull request by rotemtam**
Stacked on #245 

Adding first migrate spec types for initial review of the API. 

Starting to add types that capture the different change types from `sql/migrate`.

In HCL, it will look something like:

```hcl
modify_table {
	table = "users"
	add_column {
		column "id" {
			type = "int"
		}
	}
}
```

After building this layer, we can start exploring how the migration document looks like. For example:

```hcl
migration "xyz" {
  up {
     modify_table {
        // ...
     }
  }
  down {
     // ...
  }
}
```

### CODE REVIEW

**Review by yonidavidson — APPROVED** [review]
💯 
Just fix linter issues

File: sql/migratespec/migratespec.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package migratespec
```
Comment (a8m) [inline sql/migratespec/migratespec.go:1]: Let's keep this code in `sqlspec`?

**Review by a8m — CHANGES_REQUESTED** [review]
See my comment.

File: sql/migratespec/migratespec.go (hunk 1 above)
Comment (rotemtam) [reply sql/migratespec/migratespec.go]: done

File: sql/sqlspec/migratespec.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import (
+	"ariga.io/atlas/schema/schemaspec"
+)
+
+type (
+	// Change is the interface implemented by change specifications. Change instances are supposed
+	// to be mappable to schema.Change instances.
+	Change interface {
+		c()
```
Comment (a8m) [inline sql/sqlspec/migratespec.go:11]: https://github.com/ariga/atlas/blob/master/sql/schema/migrate.go#L24
```suggestion
		change()
```

File: internal/integration/hclsqlspec/migrate_test.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import (
+	"testing"
+
+	"ariga.io/atlas/schema/schemaspec/schemahcl"
+	"ariga.io/atlas/sql/sqlspec"
+	"github.com/stretchr/testify/require"
+)
+
+func Test(t *testing.T) {
```
Comment (a8m) [inline internal/integration/hclsqlspec/migrate_test.go:11]:
```suggestion
func TestMigrate(t *testing.T) {
```

**Review by a8m — APPROVED** [review]
Please fix the above comments.

**Pull request merged**